dpnp/dpnp_algo/dpnp_algo_arraycreation.pyx

```diff
@@ -272,6 +272,59 @@ cpdef utils.dpnp_descriptor dpnp_ones_like(result_shape, result_dtype):
     return call_fptr_1out(DPNP_FN_ONES_LIKE, utils._object_to_tuple(result_shape), result_dtype)
 
 
+cpdef dpnp_ptp(utils.dpnp_descriptor arr, axis=None):
+    cdef shape_type_c shape_arr = arr.shape
+    cdef shape_type_c output_shape
+    if axis is None:
+        axis_ = axis
+        output_shape = (1,)
+    else:
+        if isinstance(axis, int):
+            if axis < 0:
+                axis_ = tuple([arr.ndim - axis])
+            else:
+                axis_ = tuple([axis])
+        else:
+            _axis_ = []
+            for i in range(len(axis)):
+                if axis[i] < 0:
+                    _axis_.append(arr.ndim - axis[i])
+                else:
+                    _axis_.append(axis[i])
+            axis_ = tuple(_axis_)
+
+        out_shape = []
+        ind = 0
+        for id, shape_axis in enumerate(shape_arr):
+            if id not in axis_:
+                out_shape.append(shape_axis)
+        output_shape = tuple(out_shape)
+ 
+    cdef DPNPFuncType param1_type = dpnp_dtype_to_DPNPFuncType(arr.dtype)
+
+    cdef DPNPFuncData kernel_data = get_dpnp_function_ptr(DPNP_FN_PTP, param1_type, param1_type)
+
+    cdef utils.dpnp_descriptor result = utils.create_output_descriptor(output_shape, kernel_data.return_type, None)
+
+    cdef custom_arraycreation_1in_1out_func_ptr_t func = <custom_arraycreation_1in_1out_func_ptr_t > kernel_data.ptr
+    
+    cdef shape_type_c axis1
+    cdef Py_ssize_t axis_size = 0
+    cdef shape_type_c axis2 = axis1
+    if axis_ is not None:
+        axis1 = axis_
+        axis2.reserve(len(axis1))
+        for shape_it in axis1:
+            if shape_it < 0:
+                raise ValueError("DPNP dparray::__init__(): Negative values in 'shape' are not allowed")
+            axis2.push_back(shape_it)
+        axis_size = len(axis1)
+    
+    func(result.get_data(), arr.get_data(), < size_t * > output_shape.data(), result.ndim, < size_t * > shape_arr.data(), arr.ndim, < size_t * > axis2.data(), axis_size)
```

Comment:
Do you really need casting to `< size_t * >`?

Reply:
Looks like problem is in `custom_arraycreation_1in_1out_func_ptr_t`. Could you please change types `size_t *` to `long *` there?